static/src/javascripts/projects/commercial/modules/article-body-adverts.spec.js

```diff
@@ -0,0 +1,240 @@
+}));
+jest.mock('lib/detect', () => ({
+    isBreakpoint: jest.fn(),
+    getBreakpoint: jest.fn(),
+    getViewport: jest.fn(),
+}));
+jest.mock('lib/config', () => ({ page: {} }));
+
+const spaceFillerStub: JestMockFn = (spaceFiller.fillSpace: any);
+const getFirstRulesUsed = () =>
+    articleBodyAdvertsInit().then(() => {
+        const firstCall = spaceFillerStub.mock.calls[0];
+        return firstCall[0];
+    });
+
+describe('Article Body Adverts', () => {
+    beforeEach(() => {
+        jest.resetAllMocks();
+        commercialFeatures.articleBodyAdverts = true;
+        spaceFillerStub.mockImplementation(() => Promise.resolve(true));
+        detect.getViewport.mockReturnValue({ height: 1000 });
+    });
+
+    it('should exist', () => {
+        expect(articleBodyAdvertsInit).toBeDefined();
+    });
+
+    it('should exit if commercial feature disabled', () => {
+        commercialFeatures.articleBodyAdverts = false;
+        return articleBodyAdvertsInit().then(executionResult => {
+            expect(executionResult).toBe(false);
+            expect(spaceFillerStub).not.toHaveBeenCalled();
+        });
+    });
+
+    it('should call space-filler`s insertion method with the correct arguments', () =>
+        articleBodyAdvertsInit().then(() => {
+            expect(spaceFillerStub).toHaveBeenCalled();
+            const firstCallArgs = spaceFillerStub.mock.calls[0];
+            const rulesArg = firstCallArgs[0];
+            const writerArg = firstCallArgs[1];
+
+            expect(rulesArg.minAbove).toBeDefined();
+            expect(rulesArg.minBelow).toBeDefined();
+            expect(rulesArg.selectors).toBeDefined();
+
+            expect(writerArg).toEqual(expect.any(Function));
+        }));
+
+    describe('When merchandising components enabled', () => {
+        beforeEach(() => {
+            detect.getBreakpoint.mockReturnValue('mobile');
+            detect.isBreakpoint.mockReturnValue(true);
+            config.page.hasInlineMerchandise = true;
+        });
+
+        it('its first call to space-filler uses the inline-merch rules', () =>
+            articleBodyAdvertsInit().then(() => {
+                const firstCallArgs = spaceFillerStub.mock.calls[0];
+                const rules = firstCallArgs[0];
+
+                expect(rules.minAbove).toEqual(300);
+                expect(rules.selectors[' > h2'].minAbove).toEqual(100);
+            }));
+
+        it('its first call to space-filler passes an inline-merch writer', () => {
+            const fixture = document.createElement('div');
+            const paragraph = document.createElement('p');
+            fixture.appendChild(paragraph);
+
+            return articleBodyAdvertsInit().then(() => {
+                const firstCall = spaceFillerStub.mock.calls[0];
+                const writer = firstCall[1];
+                writer([paragraph]);
+
+                // Wait until fastdom has written before checking
+                return fastdom.read(() => {
```

Comment:
My feeling about this is, that we could ignore the benefits of fastdom in the specs and jsut do sync calls. It's not a real dom anyway and it introduces possible problems.

What d oyou think?